pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -698,6 +698,24 @@ protected void internalGrantPermissionOnNamespace(String role, Set<AuthAction> a
         }
     }
 
+    protected boolean getImplicitPermissionOnSubscription() {
+        validateNamespaceOperation(namespaceName, NamespaceOperation.GET_PERMISSION);
+        Policies policies = getNamespacePolicies(namespaceName);
+        return policies.auth_policies.isImplicitSubscriptionAuth();
+    }
+
+    protected void internalSetImplicitPermissionOnSubscription(boolean isImplicitPermissionOnSubscription) {
+        if (isImplicitPermissionOnSubscription) {
+            validateNamespaceOperation(namespaceName, NamespaceOperation.GRANT_PERMISSION);
+        } else {
+            validateNamespaceOperation(namespaceName, NamespaceOperation.REVOKE_PERMISSION);
+        }
+        validatePoliciesReadOnlyAccess();
+        updatePolicies(namespaceName, policies -> {
+            policies.auth_policies.setImplicitSubscriptionAuth(isImplicitPermissionOnSubscription);
+            return policies;
+        });
+    }
```

Comment:
It might be worth considering PIP-149/#14365 here and use the async style to implement the API.

To clarify it: the server side implementation should by async, but the client API can contain both sync and async API methods.